aptos-move/aptos-workspace-server/src/main.rs

```diff
@@ -99,66 +115,112 @@ async fn spawn_node(test_dir: &Path) -> Result<()> {
         }
     });
 
-    let api_port = api_port_rx.await?;
-    let indexer_grpc_port = indexer_grpc_port_rx.await?;
+    let fut_api = async move {
+        let api_port = api_port_rx.await?;
 
-    let api_health_checker = HealthChecker::NodeApi(
-        Url::parse(&format!(
-            "http://{}:{}",
-            node_config.api.address.ip(),
-            api_port
-        ))
-        .unwrap(),
-    );
-    let indexer_grpc_health_checker = HealthChecker::DataServiceGrpc(
-        Url::parse(&format!(
-            "http://{}:{}",
-            node_config.indexer_grpc.address.ip(),
-            indexer_grpc_port
-        ))
-        .unwrap(),
-    );
+        let api_health_checker = HealthChecker::NodeApi(
+            Url::parse(&format!("http://{}:{}", IP_LOCAL_HOST, api_port)).unwrap(),
+        );
+        api_health_checker.wait(None).await?;
 
-    api_health_checker.wait(None).await?;
-    eprintln!(
-        "Node API is ready. Endpoint: http://127.0.0.1:{}/",
-        api_port
-    );
+        println!(
+            "Node API is ready. Endpoint: http://{}:{}/",
+            IP_LOCAL_HOST, api_port
+        );
 
-    indexer_grpc_health_checker.wait(None).await?;
-    eprintln!(
-        "Transaction stream is ready. Endpoint: http://127.0.0.1:{}/",
-        indexer_grpc_port
-    );
+        Ok(api_port)
+    };
+
+    let fut_indexer_grpc = async move {
+        let indexer_grpc_port = indexer_grpc_port_rx.await?;
+
+        let indexer_grpc_health_checker = HealthChecker::DataServiceGrpc(
+            Url::parse(&format!("http://{}:{}", IP_LOCAL_HOST, indexer_grpc_port)).unwrap(),
+        );
+
+        indexer_grpc_health_checker.wait(None).await?;
+        println!(
+            "Transaction stream is ready. Endpoint: http://{}:{}/",
+            IP_LOCAL_HOST, indexer_grpc_port
+        );
+
+        Ok(indexer_grpc_port)
+    };
+
+    Ok((fut_api, fut_indexer_grpc))
+}
+
+/// Starts the faucet service.
+/// The port used will be returned once the service is fully up.
+async fn start_faucet(
+    test_dir: &Path,
+    fut_node_api: impl Future<Output = Result<u16, Arc<anyhow::Error>>>,
+    fut_indexer_grpc: impl Future<Output = Result<u16, Arc<anyhow::Error>>>,
+) -> Result<u16> {
+    let api_port = fut_node_api
+        .await
+        .map_err(anyhow::Error::msg)
+        .context("failed to start faucet: node api did not start successfully")?;
+
+    fut_indexer_grpc
```

Comment:
Nit, faucet only needs the node to start up first.

Reply:
This is what I thought as well, but if I don't wait for indexer grpc, I get the following error
```
Test directory: /tmp/.tmp6AShkh
Node API is ready. Endpoint: http://127.0.0.1:33883/
Faucet exited with error Failed to build Funder

Caused by:
    0: Failed to build MintFunder
    1: Failed to make MintFunder use delegated account
    2: Failed to create new account
    3: Error(44): Unknown error error sending request for url (http://127.0.0.1:33883/v1/transactions): operation timed out: []
Transaction stream is ready. Endpoint: http://127.0.0.1:37143/
Error: failed to start faucet
```
Is `v1/transactions` actually part of indexer grpc?